lib/src/main/java/com/auth0/android/lock/PasswordlessLockActivity.java

```diff
@@ -400,14 +404,14 @@ public void onCountryCodeChangeRequest(CountryCodeChangeEvent event) {
     @SuppressWarnings("unused")
     @Subscribe
     public void onPasswordlessAuthenticationRequest(PasswordlessLoginEvent event) {
-        if (configuration.getDefaultPasswordlessStrategy() == null) {
+        if (configuration.getPasswordlessConnection() == null) {
             Log.w(TAG, "There is no default Passwordless strategy to authenticate with");
             return;
         }
 
         lockView.showProgress(true);
         AuthenticationAPIClient apiClient = new AuthenticationAPIClient(options.getAccount());
-        String connectionName = configuration.getFirstConnectionOfStrategy(configuration.getDefaultPasswordlessStrategy());
+        String connectionName = configuration.getPasswordlessConnection().getName();
```

Comment:
Ok, if there is a test to avoid a subtle change in order of execution to break the assumption